code/lib/builder-vite/src/vite-config.ts

```diff
@@ -40,7 +42,22 @@ export async function commonConfig(
 ): Promise<ViteInlineConfig> {
   const configEnv = _type === 'development' ? configEnvServe : configEnvBuild;
 
-  const { config: userConfig = {} } = (await loadConfigFromFile(configEnv)) ?? {};
+  // I destructure away the `build` property from the user's config object
+  // I do this because I can contain config that breaks storybook, such as we had in a lit project.
+  // If the user needs to configure the `build` they need to do so in the viteFinal function in main.js.
+  const { config: { build: buildProperty = undefined, ...userConfig } = {} } =
+    (await loadConfigFromFile(configEnv)) ?? {};
+
+  if (buildProperty) {
+    console.warn(
+      dedent`
+        The 'build' property in your vite config is not supported by storybook. Please use the viteFinal function in main.js to configure the build.
+        This is because storybook needs to configure the build for you.
+
+        Often the user's (your) app, needs 'build' config that is different from storybook.
```

Comment:
This might be a concerning message to show to users.  Do we have any concept of `debug` logging levels?

Reply:
Yeah, I did want to add some sort of notification in case users would wonder, why this setting isn't applied.

But printing this on every start and build, when the user's has taken this into account isn't great.